### DISCUSSION TITLE
Setting new JSSE provider for Sqs Async  Client

### DISCUSSION THREAD

**Discussion by skimwpi** [aws/aws-sdk-java-v2#4204] [Migration from v1 to v2]
Hi I'm migrating from v1 to v2 and having difficulty understanding how to use BouncyCastle JSSE provider for async SQS client. In v1, we had

```kotlin

import nl.altindag.ssl.SSLFactory
import nl.altindag.ssl.util.Apache4SslUtils
import org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider
import org.bouncycastle.jsse.provider.BouncyCastleJsseProvider

  fun amazonSqsAsync(): AmazonSQSAsync {
    return AmazonSQSAsyncClientBuilder.standard()
        .withCredentials(awsCredentials())
        .withClientConfiguration(getClientConfiguration())
        .withRegion(regions())
        .build()
  }

  fun getClientConfiguration(): ClientConfiguration? {
    val socketFactory = Apache4SslUtils.toSocketFactory(sslFactory())
    val clientConfiguration = ClientConfiguration()
    clientConfiguration.apacheHttpClientConfig.withSslSocketFactory(socketFactory)
    return clientConfiguration
  }

  fun sslFactory(): SSLFactory? {
    return SSLFactory.builder()
      .withDefaultTrustMaterial()
      .withSecurityProvider(BouncyCastleJsseProvider(true, BouncyCastleFipsProvider()))
      .withProtocols(tlsVersion)
      .build()
  }
```

As the code above indicates, I'm using [sslcontext-kickstarter](https://github.com/Hakky54/sslcontext-kickstart) library (nl.altindag.ssl) to configure an Apache client with BouncyCastle JSSE provider. 

In V2, I'm not understanding how to use SdkAsyncHttpClient (Netty or CRT) to recreate the similiar code for Async client. Both AwsCrtAsyncHttpClient (CRT client) and NettyNioAsyncHttpClient (Netty Client) and their respective builders do not have an approach that I can see which would allow me to set the JSSE provider. Is the feature just not supported with SdkAsyncHttpClient or am I missing a concept here? I do see the ability to tweak TLS configurations like timeouts, etc but not to replace JSSE provider.

Versions:
AWS SDK version I'm migrating to is 2.20.103 from 1.11.1034

**Comment by debora-ito** [comment]
@skimwpi currently, custom socket factories are only supported in sync clients.

We have a feature request to support in the async clients, tracking here: https://github.com/aws/aws-sdk-java-v2/issues/2896

**Comment by skimwpi** [comment]
@debora-ito thank you for your reply, much appreciated.
